Gems/ROS2/Code/Source/Camera/CameraSensor.cpp

```diff
@@ -0,0 +1,240 @@
+            { AZ::RHI::Format::R8G8B8A8_UNORM, "rgba8" },     { AZ::RHI::Format::R16G16B16A16_UNORM, "rgba16" },
+            { AZ::RHI::Format::R32G32B32A32_FLOAT, "32FC4" }, // Unsuported by RVIZ2
+            { AZ::RHI::Format::R8_UNORM, "mono8" },           { AZ::RHI::Format::R16_UNORM, "mono16" },
+            { AZ::RHI::Format::R32_FLOAT, "32FC1" },
+        };
+
+        /// @BitDepth - contains the mapping from RHI to size used in `step` size computation.
+        /// It is some equivalent to `bitDepth()` function from `sensor_msgs/image_encodings.hpp`
+        AZStd::unordered_map<AZ::RHI::Format, int> BitDepth{
+            { AZ::RHI::Format::R8G8B8A8_UNORM, 4 * sizeof(uint8_t) },
+            { AZ::RHI::Format::R16G16B16A16_UNORM, 4 * sizeof(uint16_t) },
+            { AZ::RHI::Format::R32G32B32A32_FLOAT, 4 * sizeof(float) }, // Unsuported by RVIZ2
+            { AZ::RHI::Format::R8_UNORM, sizeof(uint8_t) },
+            { AZ::RHI::Format::R16_UNORM, sizeof(uint16_t) },
+            { AZ::RHI::Format::R32_FLOAT, sizeof(float) },
+        };
+
+    } // namespace Internal
+    CameraSensorDescription::CameraSensorDescription(const AZStd::string& cameraName, float verticalFov, int width, int height)
+        : m_verticalFieldOfViewDeg(verticalFov)
+        , m_width(width)
+        , m_height(height)
+        , m_cameraName(cameraName)
+        , m_aspectRatio(static_cast<float>(width) / static_cast<float>(height))
+        , m_viewToClipMatrix(MakeViewToClipMatrix())
+        , m_cameraIntrinsics(MakeCameraIntrinsics())
+    {
+        ValidateParameters();
+    }
+
+    AZ::Matrix4x4 CameraSensorDescription::MakeViewToClipMatrix() const
+    {
+        const float nearDist = 0.1f, farDist = 100.0f;
+        AZ::Matrix4x4 localViewToClipMatrix;
+        AZ::MakePerspectiveFovMatrixRH(
+            localViewToClipMatrix, AZ::DegToRad(m_verticalFieldOfViewDeg), m_aspectRatio, nearDist, farDist, true);
+        return localViewToClipMatrix;
+    }
+
+    void CameraSensorDescription::ValidateParameters() const
+    {
+        AZ_Assert(
+            m_verticalFieldOfViewDeg > 0.0f && m_verticalFieldOfViewDeg < 180.0f,
+            "Vertical fov should be in range 0.0 < FoV < 180.0 degrees");
+        AZ_Assert(!m_cameraName.empty(), "Camera name cannot be empty");
+    }
+
+    AZStd::array<double, 9> CameraSensorDescription::MakeCameraIntrinsics() const
+    {
+        //  Intrinsic camera matrix of the camera image is being created here
+        //  It is based on other parameters available in the structure - they must be initialized before this function is called
+        //  Matrix is row-major and has the following form:
+        //  [fx  0 cx]
+        //  [ 0 fy cy]
+        //  [ 0  0  1]
+        // Projects 3D points in the camera coordinate frame to 2D pixel
+        // coordinates using the focal lengths (fx, fy) and principal point
+        // (cx, cy).
```

Comment:
Generally, line comments '//' are risky when trying to document some form of diagram or picture. The reason is that if you use a '\' as the last line, the compiler will treat it as a continuation even if its in a comment. Even though this comment is safe, its better practice to use block comment tags instead

```suggestion
        /* Intrinsic camera matrix of the camera image is being created here
           It is based on other parameters available in the structure - they must be initialized before this function is called
            Matrix is row-major and has the following form:
            [fx  0 cx]
            [ 0 fy cy]
            [ 0  0  1]
           Projects 3D points in the camera coordinate frame to 2D pixel
           coordinates using the focal lengths (fx, fy) and principal point
           (cx, cy).  
       */
```